src/main/java/com/phonebookservice/controller/AbstractController.java

```diff
@@ -0,0 +1,33 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+
+public abstract class AbstractController {
+    private DBIf databse = null;
+
+    /**
+     * controller constructor.
+     *
+     * @param databseValue the database
+     */
+    public AbstractController(final DBIf databseValue) {
+        this.databse = databseValue;
+    }
+
+    /**
+     * create model.
+     *
+     * @param model the model
+     */
+    public abstract void create(Model model);
```

Comment:
should be final

Reply:
how come?  I cannot override final methods

Reply:
final Model model

Reply:
adding final to Model is reduntent in checkStyle